python/ql/src/experimental/Security/CWE-730/RegexInjection.ql

```diff
@@ -0,0 +1,28 @@
+/**
+ * @name Regular expression injection
+ * @description User input should not be used in regular expressions without first being escaped,
+ *              otherwise a malicious user may be able to inject an expression that could require
+ *              exponential time on certain inputs.
+ * @kind path-problem
+ * @problem.severity error
+ * @id py/regex-injection
+ * @tags security
+ *       external/cwe/cwe-730
+ *       external/cwe/cwe-400
+ */
+
+// determine precision above
+import python
+import experimental.semmle.python.security.injection.RegexInjection
+import DataFlow::PathGraph
+
+from
+  RegexInjectionFlowConfig config, DataFlow::PathNode source, DataFlow::PathNode sink,
+  Attribute sinkAttribute
+where
+  config.hasFlowPath(source, sink) and
+  sinkAttribute = sink.getNode().(RegexInjectionSink).getRegexMethod()
+select sink.getNode(), source, sink,
+  "$@ regular expression is constructed from a $@ and executed by $@.", sink.getNode(), "This",
+  source.getNode(), "user-provided value", sinkAttribute,
+  sinkAttribute.getObject().toString() + "." + sinkAttribute.getName()
```

Comment:
When it comes to `re.compile(request.args['pattern']).search("")` the actual `getObject()` is `re.compile(request.args['pattern']).search` so it gets parsed as `Attribute()search`.

Reply:
Is there any way to get the latest `Attribute` or alike @yoff?

Reply:
Another way could be `getObject().(Name).getId()` but we'd loose `re.compile().ReMethod` when casting `Name`...

Reply:
What result would you actually like? Are you really looking to report just `search` or "a call to search" here?

Reply:
`executed by re.search` for example

Reply:
But in the case of `re.compile(request.args['pattern']).search`?

Reply:
https://github.com/github/codeql/pull/5442/commits/d213881ab6a032ed296702b08486aa6271596a2d -> would say `re.search` in that case, and it's perfect 🥳